src/vcpkg/base/files.cpp

```diff
@@ -65,6 +65,9 @@ namespace
         return ntbs[0] == '.' && (ntbs[1] == '\0' || (ntbs[1] == '.' && ntbs[2] == '\0'));
     }
 #endif
+#ifdef __APPLE__
+    bool is_DS_Store_file(const dirent* entry) { return Strings::equals(entry->d_name, ".DS_Store"); }
+#endif
```

Comment:
We should move this out of the files layer, and into the places where we care about .DS_Store files.

Reply:
But we care nowhere about .DS_Store files. Otherwise we have to inject `if(is_DS_Store(entry)) continue;` into literally every for loop that iterates over a dir.

Reply:
The problem is that there are lots of places that need to consider this, things that aren't only that one binary caching place (like `nuget.exe`), etc. and doing it this way is likely to create "impossible" behavior that hurts to debug later.

Reply:
>  doing it this way is likely to create "impossible" behavior that hurts to debug later.

One can add a debug output every time a `.DS_Store` file is ignored.

Reply:
Alternatively I have created  #344